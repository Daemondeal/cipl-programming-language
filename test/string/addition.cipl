print("H" ~ "i")
print("H" + "i") # This is not valid and should give an error