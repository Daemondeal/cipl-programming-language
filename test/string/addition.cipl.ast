(expression (call print (~ H i)))
(expression (call print (+ H i)))

