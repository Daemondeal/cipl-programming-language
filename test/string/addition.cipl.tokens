Token          Lexeme         Line 
IDENTIFIER     print          1    
LEFT_PAREN     (              1    
STRING         "H"            1     H
TILDE          ~              1    
STRING         "i"            1     i
RIGHT_PAREN    )              1    
LINE_END       \n             2    
IDENTIFIER     print          2    
LEFT_PAREN     (              2    
STRING         "H"            2     H
PLUS           +              2    
STRING         "i"            2     i
RIGHT_PAREN    )              2    
LINE_END       \n             2    
EOF                           2    
