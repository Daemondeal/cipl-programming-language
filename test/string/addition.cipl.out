::stdout::
Hi

::stderr::
Error: Operands must be numbers at line 2
